crates/data_components/src/spark_connect.rs

```diff
@@ -343,12 +344,19 @@ impl ExecutionPlan for SparkConnectExecutionPlan {
         tracing::trace!("projected_schema {:#?}", self.projected_schema);
         tracing::trace!("sql columns {:#?}", columns);
         tracing::trace!("filters {:#?}", self.filters);
+        let filtered_columns = self
```

Comment:
So after reviewing - the `columns` above aren't being used by anything. Let's remove them, so we have something that looks like:

```rust
use spark_connect_rs::{
    functions::col,
    spark::{data_type, DataType},
    DataFrame, SparkSession, SparkSessionBuilder,
};
...
        let filtered_columns = self
            .projected_schema
            .fields()
            .iter()
            .map(|f| col(f.name()))
            .collect::<Vec<_>>();
        tracing::trace!("projected_schema {:#?}", self.projected_schema);
        tracing::trace!("sql columns {:#?}", filtered_columns);
        tracing::trace!("filters {:#?}", self.filters);
        let df = self
            .filters
            .iter()
            .fold(self.dataframe.clone(), |df, filter| {
                df.filter(filter.as_str())
            })
            .select(filtered_columns);
```

Reply:
That sounds good, let me change to that